clang/utils/TableGen/ClangProgModelBuiltinEmitter.cpp

```diff
@@ -848,6 +916,232 @@ void BuiltinNameEmitter::EmitQualTypeFinder() {
   OS << "\n} // Bultin2Qual\n";
 }
 
+JSONBuiltinInterfaceEmitter::TypeDesc::TypeDesc(const Record *T)
+    : Name(T->getValueAsString("Name")), VecWidth(T->getValueAsInt("VecWidth")),
+      ElementSize(T->getValueAsInt("ElementSize")),
+      IsPointer(T->getValueAsBit("IsPointer")),
+      IsConst(T->getValueAsBit("IsConst")),
+      IsVolatile(T->getValueAsBit("IsVolatile")) {
+  assert((!T->getValueAsBit("IsInteger") || !T->getValueAsBit("IsFloat")) &&
+         "A type can't be both an int and a float");
+  if (T->getValueAsBit("IsInteger")) {
+    if (ElementSize == 1)
+      ElementType = BOOL;
+    else
+      ElementType = T->getValueAsBit("IsSigned") ? INT : UINT;
+  } else {
+    if (T->getValueAsBit("IsFloat"))
+      ElementType = FLOAT;
+    else {
+      if (!T->isSubClassOf("FundamentalType") && !T->isSubClassOf("VectorType"))
+        ElementType = BUILTIN;
+      else
+        llvm_unreachable("Invalid type");
+    }
+  }
+  AddrSpace = StringSwitch<const char *>(T->getValueAsString("AddrSpace"))
```

Comment:
Make this string-switch return StringRef.  You might have to put a StringRef ctor around each of the strings, but it will save us from having to call strlen on the result.

Reply:
fixed